tools/purge_executions.py

```diff
@@ -58,101 +55,104 @@ def do_register_cli_opts(opts, ignore_errors=False):
                 raise
 
 
-def _purge_action_models(execution_db):
-    liveaction_id = execution_db.liveaction['id']
+def _register_cli_opts():
+    cli_opts = [
+        cfg.StrOpt('timestamp', default=None,
+                   help='Will delete data older than ' +
+                   'this timestamp. (default 30 days). ' +
+                   'Example value: 2015-03-13T19:01:27.255542Z'),
+        cfg.StrOpt('action-ref', default='',
+                   help='action-ref to delete executions for.')
+    ]
+    _do_register_cli_opts(cli_opts)
+
+
+def _purge_liveaction_models(execution_db):
+    liveaction_id = execution_db.liveaction.get('id', None)
 
     if not liveaction_id:
-        print('Invalid LiveAction id. Skipping delete: %s', execution_db)
+        LOG.error('Invalid LiveAction id. Skipping delete: %s', execution_db)
 
     liveaction_db = None
     try:
         liveaction_db = LiveAction.get_by_id(liveaction_id)
     except:
-        print('LiveAction with id: %s not found. Skipping delete.', liveaction_id)
+        LOG.exception('LiveAction with id: %s not found. Skipping delete.', liveaction_id)
     else:
         global DELETED_COUNT
         DELETED_COUNT += 1
 
     try:
         ActionExecution.delete(execution_db)
-    except Exception as e:
-        print('Exception deleting Execution model: %s, exception: %s',
-              execution_db, str(e))
+    except:
+        LOG.exception('Exception deleting Execution model: %s',
+                      execution_db)
     else:
-        try:
-            LiveAction.delete(liveaction_db)
-        except Exception as e:
-            print('Zombie LiveAction left in db: %s. Exception: %s',
-                  liveaction_db, str(e))
+        if liveaction_db:
+            try:
+                LiveAction.delete(liveaction_db)
+            except:
+                LOG.exception('Zombie LiveAction left in db: %s. Exception: %s',
+                              liveaction_db)
+
 
+def _should_delete(execution_db, action_ref, timestamp):
+    if execution_db.status in IN_PROGRESS_STATES:
+        return False
```

Comment:
> @manasdk @Kami Need some opinions here. I made a choice that if executions older than cut off timestamp are not in DONE states (succeeded, failed, timedout, canceled), I leave them in the DB. What do you guys think? I have no strong opinions here.

I think that is a reasonable default, but also need to take into consideration what phool and manasdk say (looks like it's already done).